WalletWasabi/Extensions/LinqExtensions.cs

```diff
@@ -193,5 +193,19 @@ public static IEnumerable<(T1, T2)> ZipForceEqualLength<T1, T2>(this IEnumerable
 			}
 			return source.Zip(otherCollection);
 		}
+
+		public static IEnumerable<TAccumulate> Scan<TSource, TAccumulate>(
+			this IEnumerable<TSource> source,
+			TAccumulate seed,
+			Func<TAccumulate, TSource, TAccumulate> func)
+		{
+			TAccumulate previous = seed;
+			foreach (var item in source)
+			{
+				var result = func(previous, item);
+				previous = result;
+				yield return result;
```

Comment:
it looks to me that `result` variable is not necessary at all:

```suggestion
                              previous = func(previous, item);
				yield return previous;
```

Reply:
Yes, this is right. I didn't do it in that way because the `return previous` is confusing but it seems in this way is how we devs expect to see it.

Reply:
Done.